Assets/HoloToolkit/Input/Scripts/Cursor/AnimatedCursor.cs

```diff
@@ -0,0 +1,157 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+using System;
+using UnityEngine;
+
+namespace HoloToolkit.Unity.InputModule
+{
+    /// <summary>
+    /// Animated cursor is a cursor driven using an animator to inject state information
+    /// and animate accordingly
+    /// </summary>
+    public class AnimatedCursor : Cursor
+    {
+        /// <summary>
+        /// Data struct for cursor state information for the Animated Cursor
+        /// </summary>
+        [Serializable]
+        public struct AnimCursorDatum
+        {
+            public string Name;
+            public CursorStateEnum CursorState;
+
+            public enum AnimInputTypeEnum
```

Comment:
You mean the enum values for AnimInputTypeEnum? They just match the [Unity types for Unity animations](https://docs.unity3d.com/Manual/AnimationParameters.html).

I guess it might not be obvious for people that haven't used Unity's animations. I can add a comment on the enum if you want.